### DISCUSSION TITLE
GC9A01 Ring Meter

### DISCUSSION THREAD

**Discussion by morego815** [Bodmer/TFT_eSPI#1883] [Q&A - General] [answered]
I've been trying to get the [ring meter project](https://www.instructables.com/Arduino-analogue-ring-meter-on-colour-TFT-display/) to work on a GC9A01 display with Arduino Nano but I haven't been able to find a combination of drivers and libraries to get it working.  Will this library or maybe a slightly modified version work?

**Comment by Bodmer** [comment, marked as answer]
The Arduino Nano does not have suffficient RAM, so will not work with the library. The library is aimed at more powerful 32 bit processor.
Many GC9A01 displays cannot be read and thus will not run correctly with that sketch.

**Reply by morego815** [reply]
I've had little success trying to modify the ILI9341_AS driver using the Adafruit GC9A01 library for reference.  I can't get it to show on the display, but the ring meter does display for a second when I flash back to the working Adafruit graphics test sketch.  Gives me hope that I'm just missing something somewhere and it will work if I find it.

**Reply by Bodmer** [reply]
Ah, This is the TFT_eSPI repository.  It is difficult to provide advice on library modifications due to the number of potential pitfalls.

**Reply by morego815** [reply]
Yeah I found this repository and The Adafruit CG9A01 driver uses the Adafruit SPITFT library so I thought maybe I could use it or some parts of it to get the ring meter working.

**Reply by pkathey** [reply]
Hello there and thank you very much Bodmer for all your excellent work!

I recently acquired a Waveshare 1.28 inch GC9A01-based LCD display and managed to adapt your Arduino 'ring' meter code for my purposes using a Raspberry Pico ...seems to work OK in terms of my test code though I am a complete novice at this. I have yet to use with a real temperature probe as input. Regardless, I have included a copy in case it may be of use.

[GC9A01_Temp_Meter.ino.zip](https://github.com/Bodmer/TFT_eSPI/files/9217610/GC9A01_Temp_Meter.ino.zip)
